### ISSUE TITLE
SWMR tests failing

### ISSUE THREAD

**Issue by ArchangeGabriel** [h5py/h5py#890]
Environment:
 * ArchLinux
 * Python 2.7+3.6 (system)
 * h5py version (e.g. 2.7)
 * HDF5 version (e.g. 1.10.0) (with Fortran, MPI, CPP, Parallel)

After running `python setup.py configure --mpi` and `python setup.py build`, I’ve got failure with `python setup.py test`:
```
running test
running build_py
running build_ext
Autodetected HDF5 1.10.0
********************************************************************************
                       Summary of the h5py configuration

    Path to HDF5: None
    HDF5 Version: '1.10.0'
     MPI Enabled: True
Rebuild Required: False

********************************************************************************
Executing cythonize()
/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/h5p.pyx: cannot find cimported module 'mpi4py.mpi_c'
.....................................................x.........................s.........................................x....................................s.............................................................................................................ssssss.................................................................x....x........................x.....x......................................EEEEEEE.EEssss...........
======================================================================
ERROR: test_force_swmr_mode_off_raises (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrRead)
Switching SWMR write mode off is only possible by closing the file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 58, in setUp
    self.f = h5py.File(fname, 'r', swmr=True)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 271, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 100, in make_fid
    flags |= h5f.ACC_SWMR_READ
AttributeError: module 'h5py.h5f' has no attribute 'ACC_SWMR_READ'

======================================================================
ERROR: test_force_swmr_mode_on_raises (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrRead)
Verify when reading a file cannot be forcibly switched to swmr mode.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 58, in setUp
    self.f = h5py.File(fname, 'r', swmr=True)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 271, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 100, in make_fid
    flags |= h5f.ACC_SWMR_READ
AttributeError: module 'h5py.h5f' has no attribute 'ACC_SWMR_READ'

======================================================================
ERROR: test_initial_swmr_mode_on (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrRead)
Verify that the file is initially in SWMR mode
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 58, in setUp
    self.f = h5py.File(fname, 'r', swmr=True)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 271, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 100, in make_fid
    flags |= h5f.ACC_SWMR_READ
AttributeError: module 'h5py.h5f' has no attribute 'ACC_SWMR_READ'

======================================================================
ERROR: test_read_data (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrRead)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 58, in setUp
    self.f = h5py.File(fname, 'r', swmr=True)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 271, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 100, in make_fid
    flags |= h5f.ACC_SWMR_READ
AttributeError: module 'h5py.h5f' has no attribute 'ACC_SWMR_READ'

======================================================================
ERROR: test_refresh (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrRead)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 58, in setUp
    self.f = h5py.File(fname, 'r', swmr=True)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 271, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 100, in make_fid
    flags |= h5f.ACC_SWMR_READ
AttributeError: module 'h5py.h5f' has no attribute 'ACC_SWMR_READ'

======================================================================
ERROR: test_extend_dset (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrWrite)
Extend and flush a SWMR dataset
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 127, in test_extend_dset
    self.f.swmr_mode = True
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3179)
    with _phil:
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3137)
    return func(*args, **kwds)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 227, in swmr_mode
    self.id.start_swmr_write()
AttributeError: 'h5py.h5f.FileID' object has no attribute 'start_swmr_write'

======================================================================
ERROR: test_extend_dset_multiple (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrWrite)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 140, in test_extend_dset_multiple
    self.f.swmr_mode = True
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3179)
    with _phil:
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3137)
    return func(*args, **kwds)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 227, in swmr_mode
    self.id.start_swmr_write()
AttributeError: 'h5py.h5f.FileID' object has no attribute 'start_swmr_write'

======================================================================
ERROR: test_switch_swmr_mode_off_raises (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrWrite)
Switching SWMR write mode off is only possible by closing the file.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 118, in test_switch_swmr_mode_off_raises
    self.f.swmr_mode = True
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3179)
    with _phil:
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3137)
    return func(*args, **kwds)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 227, in swmr_mode
    self.id.start_swmr_write()
AttributeError: 'h5py.h5f.FileID' object has no attribute 'start_swmr_write'

======================================================================
ERROR: test_switch_swmr_mode_on (h5py.tests.hl.test_dataset_swmr.TestDatasetSwmrWrite)
Switch to SWMR mode and verify
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/tests/hl/test_dataset_swmr.py", line 111, in test_switch_swmr_mode_on
    self.f.swmr_mode = True
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3179)
    with _phil:
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper (/<pathto>python-h5py-openmpi/src/h5py-2.7.0/h5py/_objects.c:3137)
    return func(*args, **kwds)
  File "/<pathto>python-h5py-openmpi/src/h5py-2.7.0/build/lib.linux-x86_64-3.6/h5py/_hl/files.py", line 227, in swmr_mode
    self.id.start_swmr_write()
AttributeError: 'h5py.h5f.FileID' object has no attribute 'start_swmr_write'

----------------------------------------------------------------------
Ran 439 tests in 0.518s

FAILED (errors=9, skipped=12, expected failures=6)
```

Is that expected?

**Comment by ArchangeGabriel** [comment]
Retried in a clean environment and it worked, not sure about the diff but I consider this solved. Sorry for the noise.

**Comment by tacaswell** [comment]
I suspect you had stale cython output from a hdf5 1.8 build.

**Comment by mike-lawrence** [comment]
I encountered something similar, ended up deleting /usr/local, reinstalling hdf5 1.10.1_2 & python via brew (I'm on OS X), only to find that both pip-installed and source-installed h5py thought it was using hdf5 1.8. Of note is that when I installed from source, it reported finding hdf5 1.10 then after install `h5py.version.info` reported using hdf5 1.8. Also of note is that I couldn't find hdf5 1.8 anywhere on my system (using `find / -name *hdf5*`).

I finally got it to work and I *think* that installing via `pip2 install h5py --verbose --no-cache-dir --no-binary :all:` was the trick.
